types/document.d.ts

```diff
@@ -186,10 +186,8 @@ declare module 'mongoose' {
     $parent(): Document | undefined;
 
     /** Populates document references. */
-    populate<Paths = {}>(path: string | PopulateOptions | (string | PopulateOptions)[], callback: Callback<this & Paths>): void;
-    populate<Paths = {}>(path: string, names: string, callback: Callback<this & Paths>): void;
```

Comment:
Don't get rid of these. The issue is that, if there's more than 1 non-callback arg, `path` must be a string. We don't support `populate(PopulateOptions, string, Model<unknown>)`, for example. So keep this syntax.